tests/integration.rs

```diff
@@ -13,63 +14,74 @@ mod integration {
     use ln_types::P2PAddress;
     use loin::scheduler;
     use loin::{scheduler::{ScheduledChannel, ScheduledPayJoin, Scheduler}, lnd::LndClient, http};
-    use std::collections::HashMap;
-    use std::convert::TryFrom;
     use tempfile::tempdir;
     use tonic_lnd::rpc::{ConnectPeerRequest, LightningAddress};
     use hyper_tls::HttpsConnector;
     use tokio_native_tls::native_tls;
 
+    fn do_event() -> Result<bitcoincore_rpc::Client, bitcoincore_rpc::Error> {
+        Client::new("http://localhost:43782", Auth::UserPass("ceiwHEbqWI83".to_string(), "DwubwWsoo3".to_string()))
+    }
     #[tokio::test]
     async fn test() -> Result<(), Box<dyn std::error::Error>> {
         let localhost = vec!["localhost".to_string()];
         let cert = rcgen::generate_simple_self_signed(localhost)?;
         let ssl_dir = format!("{}/tests/compose/nginx/ssl", env!("CARGO_MANIFEST_DIR"));
-        std::fs::write(format!("{}/localhost-key.pem", ssl_dir), cert.serialize_private_key_pem()).expect("unable to write file");
-        std::fs::write(format!("{}/localhost.pem", ssl_dir), cert.serialize_pem()?).expect("unable to write file");
+        std::fs::write(format!("{}/localhost-key.pem", ssl_dir), cert.serialize_private_key_pem())
+            .expect("unable to write file");
+        std::fs::write(format!("{}/localhost.pem", ssl_dir), cert.serialize_pem()?)
+            .expect("unable to write file");
 
         let compose_dir = format!("{}/tests/compose", env!("CARGO_MANIFEST_DIR"));
         let fixture = Fixture::new(compose_dir);
         let tmp_path = fixture.tmp_path();
 
         // wait for bitcoind to start and for lnd to be fully initialized with secrets
-        std::thread::sleep(std::time::Duration::from_secs(10));
 
         // sanity check
-        let bitcoin_rpc = Client::new(
-            "http://localhost:43782",
-            Auth::UserPass("ceiwHEbqWI83".to_string(), "DwubwWsoo3".to_string()),
-        )
-        .unwrap();
-        assert!(&bitcoin_rpc.get_best_block_hash().is_ok());
-
-        Command::new("docker")
-            .arg("cp")
-            .arg("compose-merchant_lnd-1:/root/.lnd/tls.cert")
-            .arg(format!("{}/merchant-tls.cert", tmp_path))
-            .output()
-            .expect("failed to copy tls.cert");
-        println!("copied merchant-tls.cert");
-
-        Command::new("docker")
-            .arg("cp")
-            .arg("compose-merchant_lnd-1:/data/chain/bitcoin/regtest/admin.macaroon")
-            .arg(format!("{}/merchant-admin.macaroon", &tmp_path))
-            .output()
-            .expect("failed to copy admin.macaroon");
-        println!("copied merchant-admin.macaroon");
+        let bitcoin_rpc = loop {
+            if let Ok(btcrpc) = do_event() {
+                if btcrpc.get_best_block_hash().is_ok() {
```

Comment:
Infinite loop got me here, changing to 
```
match btcrpc.get_best_block_hash() {
    Ok(_) => break btcrpc,
    Err(e) => println!("Attempting to contact btcrpc: {}", e),
}
```
would be informative of where the test is breaking